src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveWithVariousMembersTests.cs

```diff
@@ -153,6 +150,51 @@ public void AddInterfaceImplementationTest()
                 Assert.NotNull(testType.GetNestedType("NestedType", BindingFlags.NonPublic));
             }
         }
+
+        [Fact]
+        public void SaveGenericTypeParametersForAType()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                AssemblyBuilder assemblyBuilder = AssemblyTools.PopulateAssemblyBuilderAndSaveMethod(
+                    s_assemblyName, null, typeof(string), out MethodInfo saveMethod);
+                ModuleBuilder mb = assemblyBuilder.DefineDynamicModule("My Module");
+                TypeBuilder tb = mb.DefineType("TestInterface", TypeAttributes.Interface | TypeAttributes.Abstract);
+                string[] typeParamNames = new string[] { "TFirst", "TSecond", "TThird" };
+                GenericTypeParameterBuilder[] typeParams = tb.DefineGenericParameters(typeParamNames);
+                typeParams[0].SetInterfaceConstraints(new Type[] { typeof(IAccess), typeof(INoMethod)});
+                typeParams[1].SetCustomAttribute(new CustomAttributeBuilder(typeof(DynamicallyAccessedMembersAttribute).GetConstructor(
+                    new Type[] { typeof(DynamicallyAccessedMemberTypes) }), new object[] { DynamicallyAccessedMemberTypes.PublicProperties }));
+                typeParams[2].SetBaseTypeConstraint(typeof(EmptyTestClass));
+                typeParams[2].SetGenericParameterAttributes(GenericParameterAttributes.ReferenceTypeConstraint);
+                saveMethod.Invoke(assemblyBuilder, new object[] { file.Path });
+                // internal interface TestInterface<TFirst, [DynamicallyAccessedMembers(/*Could not decode attribute arguments.*/)] TSecond, TThird>
+                // where TFirst : IAccess, INoMethod where TThird : class, EmptyTestClass
```

Comment:
This is how it looks in the saved assembly (with ILSpy)

Reply:
Is that a bug in ILSpy or bug in RefEmit?

If it is a bug in ILSpy, we should not be including the error text in the comment.

Reply:
You can check whether ILSpy fails in the same way when this code is compiled by Roslyn.

Reply:
> Is that a bug in ILSpy or bug in RefEmit?

That is what I am trying to clarify 😄, there is 2 issue in this IL:
- `/*Could not decode attribute arguments.*/` part is probably ILSpy issue as the constructor argument was successfully read from file in this unit test. 
- The most questionable part is `class` constraint in `where TThird : class, EmptyTestClass`
I guess this is not expected, assume it should be just `where TThird : EmptyTestClass`, not sure how `class` constraint added

Reply:
Ah, my bad 😛, was added `typeParams[2].SetGenericParameterAttributes(GenericParameterAttributes.ReferenceTypeConstraint);` for testing which is causing the `class` constraint, so no issue here, removed the comment